vivisect/impemu/platarch/arm.py

```diff
@@ -110,6 +110,8 @@ def _prep(self, funcva, tmode=None):
                         armthumb -= 5
                     elif thumbop.mnem == 'ldr':
                         armthumb -= 2
+                    elif thumbop.mnem == 'b':
```

Comment:
more heuristics to make ARM/Thumb detection better.

Reply:
Can you give me a rundown of how this makes detection better and where `-= 2` comes from? Mostly for my own curiosity.

Reply:
this allows for Branches to be a likely Thumb instruction to begin a function (which actually happens in some of our test-bins) as somewhat likely, but not as likely as `ldr`.  if `armthumb` is negative, that indicates that it is most likely THUMB.  if positive, it is believed to be ARM.  this is the fallback case if we don't have some other indicator (like the instruction being not 4-byte-aligned or the workspace having a location which specifies).
this was the original fix i was attempting, when i discovered that ARM/THUMB info wasn't being communicated to makeFunction.  so once i fixed this, i would see that THUMB was being correctly-identifiee